src/commands/Pokemon/pokedex.ts

```diff
@@ -0,0 +1,169 @@
+				if (poke.preevolutions[0].preevolutions?.length) {
+					evoChain = this.parseEvoChain(poke.preevolutions[0].preevolutions[0].species, null, evoChain, false);
+				}
+			}
+
+			// Parse evolution chain and add to evochain
+			if (poke.evolutions?.length) {
+				evoChain = this.parseEvoChain(poke.evolutions[0].species, poke.evolutions[0].evolutionLevel, evoChain);
+
+				// In case there are multiple evolutionary paths
+				const otherFormeEvos = poke.evolutions.slice(1);
+				if (otherFormeEvos.length) {
+					evoChain = `${evoChain}, ${otherFormeEvos.map(oevo => `\`${oevo.species}\` (${oevo.evolutionLevel})`).join(', ')}`;
+				}
+
+				// If the direct evolution has another evolution (charmander -> charmeleon -> charizard)
+				if (poke.evolutions[0].evolutions?.length) {
+					evoChain = this.parseEvoChain(poke.evolutions[0].evolutions[0].species, poke.evolutions[0].evolutions[0].evolutionLevel, evoChain);
+				}
+			}
+
+			const embedTranslations = message.language.tget('COMMAND_POKEDEX_EMBED_DATA');
+			return message.sendEmbed(new MessageEmbed()
+				.setColor(this.resolveColour(poke.color))
+				.setAuthor(`#${poke.num} - ${toTitleCase(poke.species)}`, POKEMON_EMBED_THUMBNAIL)
+				.setThumbnail(message.flagArgs.shiny ? poke.shinySprite : poke.sprite)
+				.addField(embedTranslations.TYPES, poke.types.join(', '), true)
+				.addField(embedTranslations.ABILITIES, abilities.join(', '), true)
+				.addField(embedTranslations.GENDER_RATIO, this.parseGenderRatio(poke.gender), true)
+				.addField(embedTranslations.SMOGON_TIER, poke.smogonTier, true)
+				.addField(embedTranslations.HEIGHT, `${poke.height}m`, true)
+				.addField(embedTranslations.WEIGHT, `${poke.weight}kg`, true)
+				.addField(embedTranslations.EGG_GROUPS, poke.eggGroups?.join(', ') || '', true)
+				.addBlankField(true)
+				.addField(embedTranslations.OTHER_FORMES, poke.otherFormes?.join(', ') || embedTranslations.NONE, true)
+				.addField(embedTranslations.EVOLUTIONARY_LINE, evoChain)
+				.addField(embedTranslations.BASE_STATS, baseStats.join(', '))
+				.addField(embedTranslations.FLAVOUR_TEXT, `\`(${poke.flavorTexts[0].game})\` ${poke.flavorTexts[0].flavor}`)
+				.addField(embedTranslations.EXTERNAL_RESOURCES, [
+					`[Bulbapedia](${parseBulbapediaURL(poke.bulbapediaPage)} )`,
+					`[Serebii](${poke.serebiiPage})`,
+					`[Smogon](${poke.smogonPage})`
+				].join(' | ')));
+		} catch (err) {
+			throw message.language.tget('COMMAND_POKEDEX_QUERY_FAIL', pokemon);
+		}
+	}
+
+	private async fetchAPI(message: KlasaMessage, pokemon: string) {
+		try {
+			return await fetch(POKEMON_GRAPHQL_API_URL, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json'
+				},
+				body: JSON.stringify({
+					query: getPokemonDetailsByFuzzy(pokemon)
+				})
+			}, 'json') as Promise<GraphQLPokemonResponse<'getPokemonDetailsByFuzzy'>>;
+		} catch (err) {
+			throw message.language.tget('SYSTEM_QUERY_FAIL');
+		}
+	}
+
+	private resolveColour(col: string) {
```

Comment:
This could probably use an enum in case we use it later

Reply:
The colours it switch-cases are the colours are officially defined by the Pokedex so I disagree.

Reply:
I meant a string enum for colors

Reply:
Can't be done. Given this enum:
```ts
enum PokemonColours {
	'Black' = 0x323232,
	'Blue' = 0x257CFF,
	'Brown' = 0xA3501A,
	'Gray' = 0x969696,
	'Green' = 0x3EFF4E,
	'Pink' = 0xFF65A5,
	'Purple' = 0xA63DE8,
	'Red' = 0xFF3232,
	'White' = 0xE1E1E1,
	'Yellow' = 0xFFF359,
}
```

Results in this error:
![image](https://user-images.githubusercontent.com/4019718/69000354-83752880-08ce-11ea-899c-839fb7655745.png)


Making it `const enum` is no good either because const enums can only be accessed through dot notation.
Casting `poke.colors` is impossible as well.